crawl-ref/source/beam.cc

```diff
@@ -5202,6 +5204,23 @@ void bolt::affect_monster(monster* mon)
 
     hit_count[mon->mid]++;
 
+    // Jiyva absorbs attacks on slimes.
+    if (!is_tracer && agent()->is_player()
```

Comment:
In the arena, agent() can return NULL, so checking agent()->is_player() here will segfault if you don't check agent() first (which is why some of the tests are failing).

Reply:
Thanks. Should be fixed now.